#  Statistical Analysis & Hypothesis Testing
## Research Question
**What drives global news coverage?** Is it Military Power, Economic Wealth, or Democratic Status?

In this notebook, we visualize the relationships and perform statistical tests to determine the strongest predictor of news frequency.

#### **GDP**

**Spearman T**

In [11]:
import pandas as pd
from scipy.stats import spearmanr

# --- Configuration ---
# Set the significance level (standard alpha value)
ALPHA = 0.05 

# 1. Load the final data file
df = pd.read_csv('final_complete_dataset.csv')

# 2. Select the two variables for correlation
gdp = df['Average_GDP_Last_10_Years_Billion_USD']
frequency = df['Frequency']

# 3. Perform Spearman's Rank Correlation Test
rho, p_value = spearmanr(gdp, frequency)

# --- Output and Rejection Decision ---
print(f"--- Spearman's Rank Correlation Test Results ---")
print(f"Correlation Coefficient (Rho): {rho:.3f}")
print(f"P-value: {p_value:.5f}")
print(f"Significance Level (Alpha): {ALPHA}")
print("-" * 50)

# 4. Check for rejection of the Null Hypothesis
if p_value < ALPHA:
    rejection_statement = "P-value is less than Alpha."
    rejection_decision = "Decision: REJECT the Null Hypothesis (H0)."
    conclusion = "Conclusion: There IS a statistically significant relationship between GDP and news Frequency."
else:
    rejection_statement = "P-value is greater than Alpha."
    rejection_decision = "Decision: FAIL TO REJECT the Null Hypothesis (H0)."
    conclusion = "Conclusion: We do not have sufficient evidence to claim a relationship between GDP and news Frequency."

print(rejection_statement)
print(rejection_decision)
print(conclusion)

--- Spearman's Rank Correlation Test Results ---
Correlation Coefficient (Rho): 0.610
P-value: 0.00000
Significance Level (Alpha): 0.05
--------------------------------------------------
P-value is less than Alpha.
Decision: REJECT the Null Hypothesis (H0).
Conclusion: There IS a statistically significant relationship between GDP and news Frequency.


**Two Sample T**

In [10]:
import pandas as pd
from scipy.stats import ttest_ind

# 1. Load the data
df = pd.read_csv('final_complete_dataset.csv')

# 2. Sort the data by GDP to prepare for the split
df_sorted = df.sort_values(by='Average_GDP_Last_10_Years_Billion_USD', ascending=False).reset_index(drop=True)

# 3. Split the sorted data exactly in half (median split)
N = len(df_sorted)
midpoint = N // 2

group_high_gdp_freq = df_sorted.head(midpoint)['Frequency']
group_low_gdp_freq = df_sorted.tail(midpoint)['Frequency']

# 4. Perform Welch's t-test (Difference of Means)
t_statistic, p_value = ttest_ind(group_high_gdp_freq, 
                                 group_low_gdp_freq, 
                                 equal_var=False)

# 5. --- CALCULATE AND PRINT DEGREES OF FREEDOM (DF) ---
n1 = len(group_high_gdp_freq)
n2 = len(group_low_gdp_freq)

# Calculate Sample Variances (s^2)
s1_sq = group_high_gdp_freq.var(ddof=1)
s2_sq = group_low_gdp_freq.var(ddof=1)

# Calculate the components of the Welch–Satterthwaite equation
numerator = (s1_sq / n1 + s2_sq / n2)**2
denominator = (s1_sq / n1)**2 / (n1 - 1) + (s2_sq / n2)**2 / (n2 - 1)

df_welch = numerator / denominator
# --------------------------------------------------------

# 6. Output the results
print("\n" + "=" * 50)
print("  RESULTS OF WELCH'S T-TEST (HIGH GDP vs LOW GDP)  ")
print("=" * 50)
print(f"Group Sizes: High GDP (n={n1}), Low GDP (n={n2})")
print(f"Mean Frequency - High GDP: {group_high_gdp_freq.mean():.2f}")
print(f"Mean Frequency - Low GDP: {group_low_gdp_freq.mean():.2f}")
print("-" * 50)
print(f"T-Statistic: {t_statistic:.3f}")
print(f"P-value: {p_value:.5f}")
print(f"Degrees of Freedom (df): {df_welch:.3f}") # Added print statement
print("=" * 50)
if p_value < ALPHA:
    rejection_statement = "P-value is less than Alpha."
    rejection_decision = "Decision: REJECT the Null Hypothesis (H0)."
    conclusion = "Conclusion: There IS a statistically significant relationship between GDP and news Frequency."
else:
    rejection_statement = "P-value is greater than Alpha."
    rejection_decision = "Decision: FAIL TO REJECT the Null Hypothesis (H0)."
    conclusion = "Conclusion: We do not have sufficient evidence to claim a relationship between GDP and news Frequency."

print(rejection_statement)
print(rejection_decision)
print(conclusion)


  RESULTS OF WELCH'S T-TEST (HIGH GDP vs LOW GDP)  
Group Sizes: High GDP (n=72), Low GDP (n=72)
Mean Frequency - High GDP: 25005.78
Mean Frequency - Low GDP: 9735.21
--------------------------------------------------
T-Statistic: 4.881
P-value: 0.00000
Degrees of Freedom (df): 89.741
P-value is less than Alpha.
Decision: REJECT the Null Hypothesis (H0).
Conclusion: There IS a statistically significant relationship between GDP and news Frequency.


#### **DEMOCRACY**

**Spearman**

In [7]:
import pandas as pd
from scipy.stats import spearmanr

# 1. Load the data
df = pd.read_csv('final_complete_dataset.csv')

# 2. Select variables and handle missing values to get the true sample size (n)
df_corr = df[['Average_Democracy_Score', 'Frequency']].dropna()

democracy = df_corr['Average_Democracy_Score']
frequency = df_corr['Frequency']

# 3. Calculate Sample Size (n) and Degrees of Freedom (df)
n = len(df_corr)
# df = n - 2 is used for the t-approximation for correlation significance
df_spearman = n - 2

# 4. Perform Spearman's Rank Correlation Test
rho, p_value_rho = spearmanr(democracy, frequency)

# 5. Output the results



print("[TEST 1: SPEARMAN'S RANK CORRELATION]")
print("=" * 60)
print(f"Sample Size (n): {n}")
print(f"Degrees of Freedom (df): {df_spearman}")
print("-" * 50)
print(f"Correlation Coefficient (Rho): {rho:.3f}")
print(f"P-value: {p_value_rho:.5f}")
print("=" * 60)

[TEST 1: SPEARMAN'S RANK CORRELATION]
Sample Size (n): 145
Degrees of Freedom (df): 143
--------------------------------------------------
Correlation Coefficient (Rho): 0.292
P-value: 0.00037


**Two Sample T**

In [13]:
# --- TEST 2: DIFFERENCE OF MEANS (Democracy Split vs. Frequency) ---

# 2a. Sort the data by Democracy Score
df_sorted = df.sort_values(by='Average_Democracy_Score', ascending=False).reset_index(drop=True)

# 2b. Split the data in half (median split)
N = len(df_sorted)
midpoint = N // 2
group_high_demo_freq = df_sorted.head(midpoint)['Frequency']
group_low_demo_freq = df_sorted.tail(midpoint)['Frequency']

# 2c. Perform Welch's t-test
t_statistic, p_value_t = ttest_ind(group_high_demo_freq, 
                                    group_low_demo_freq, 
                                    equal_var=False)

# 2d. Calculate Degrees of Freedom (df) for Welch's test
n1, n2 = len(group_high_demo_freq), len(group_low_demo_freq)
s1_sq = group_high_demo_freq.var(ddof=1)
s2_sq = group_low_demo_freq.var(ddof=1)
numerator = (s1_sq / n1 + s2_sq / n2)**2
denominator = (s1_sq / n1)**2 / (n1 - 1) + (s2_sq / n2)**2 / (n2 - 1)
df_welch = numerator / denominator
print("\n[TEST 2: DIFFERENCE OF MEANS (Welch's t-test)]")
print(f"Mean Frequency - High Democracy Group: {group_high_demo_freq.mean():.2f}")
print(f"Mean Frequency - Low Democracy Group: {group_low_demo_freq.mean():.2f}")
print("-" * 50)
print(f"T-Statistic: {t_statistic:.3f}")
print(f"P-value: {p_value_t:.5f}")
print(f"Degrees of Freedom (df): {df_welch:.3f}")
print("=" * 60)


[TEST 2: DIFFERENCE OF MEANS (Welch's t-test)]
Mean Frequency - High Democracy Group: 18748.29
Mean Frequency - Low Democracy Group: 12633.95
--------------------------------------------------
T-Statistic: 2.052
P-value: 0.04189
Degrees of Freedom (df): 150.705


#### **MILITARY**

**Spearman**

In [4]:
import pandas as pd
import scipy.stats as stats
import os


df = pd.read_csv("final_complete_dataset.csv")

# 2. Prepare Data (Higher Score = Stronger Military)
# We invert PowerIndex because normally 0.0 is perfect strength.
df['Military_Strength'] = 1 / (df['PowerIndex'] + 0.0001)

# 3. Run Spearman Correlation Test
corr, p_value = stats.spearmanr(df['Military_Strength'], df['Frequency'])

print("--- Spearman Correlation Test Results ---")
print(f"Correlation Coefficient: {corr:.3f}")
print(f"P-Value: {p_value:.3e}")

if p_value < 0.05:
    print("CONCLUSION: Statistically SIGNIFICANT relationship.")
    print("Countries with stronger militaries are mentioned more frequently.")
else:
    print("CONCLUSION: No significant relationship found.")

--- Spearman Correlation Test Results ---
Correlation Coefficient: 0.654
P-Value: 4.879e-19
CONCLUSION: Statistically SIGNIFICANT relationship.
Countries with stronger militaries are mentioned more frequently.


**Two Sample T**

In [6]:
import pandas as pd
import scipy.stats as stats
import os


df = pd.read_csv("final_complete_dataset.csv")

# 2. Prepare Data
df['Military_Strength'] = 1 / (df['PowerIndex'] + 0.0001)

# 3. Split into Two Groups (High vs. Low)
median_strength = df['Military_Strength'].median()

group_strong = df[df['Military_Strength'] >= median_strength]['Frequency']
group_weak = df[df['Military_Strength'] < median_strength]['Frequency']

# 4. Run T-Test (Independent Samples)
# We use equal_var=False (Welch's t-test) because variance often differs between groups
t_stat, p_value = stats.ttest_ind(group_strong, group_weak, equal_var=False)

print("--- Two-Sample T-Test Results ---")
print(f"Median Split Point: {median_strength:.2f}")
print(f"Mean Frequency (Strong Group): {group_strong.mean():,.0f}")
print(f"Mean Frequency (Weak Group):   {group_weak.mean():,.0f}")
print(f"T-Statistic: {t_stat:.3f}")
print(f"P-Value: {p_value:.3e}")

if p_value < 0.05:
    print("CONCLUSION: Statistically SIGNIFICANT difference.")
    print("The 'Strong Military' group receives significantly more coverage.")
else:
    print("CONCLUSION: No significant difference between the groups.")

--- Two-Sample T-Test Results ---
Median Split Point: 0.72
Mean Frequency (Strong Group): 25,085
Mean Frequency (Weak Group):   9,370
T-Statistic: 5.104
P-Value: 1.729e-06
CONCLUSION: Statistically SIGNIFICANT difference.
The 'Strong Military' group receives significantly more coverage.


#### **FSI**

**Spearman**

In [2]:
from scipy.stats import spearmanr
import pandas as pd

# 1. Load Data (if not already loaded in previous cells)
df = pd.read_csv('final_complete_dataset.csv')

# 2. Filter Data
df_clean = df.dropna(subset=['Average_FSI_Total_Score', 'Frequency'])

# 3. Perform Spearman's Rank Correlation
rho, p_value = spearmanr(df_clean['Average_FSI_Total_Score'], df_clean['Frequency'])

# 4. Output Results
print("--- Spearman's Rank Correlation Test Results (FSI) ---")
print(f"Correlation Coefficient (Rho): {rho:.3f}")
print(f"P-value: {p_value:.5f}")

if p_value < 0.05:
    print("CONCLUSION: Statistically SIGNIFICANT correlation.")
else:
    print("CONCLUSION: No significant correlation.")

--- Spearman's Rank Correlation Test Results (FSI) ---
Correlation Coefficient (Rho): -0.338
P-value: 0.00004
CONCLUSION: Statistically SIGNIFICANT correlation.


**Two-Sample T**

In [4]:
from scipy.stats import ttest_ind
import pandas as pd

# 1. Prepare Data for Split
# Sort by FSI Score (Descending: Higher Score = More Fragile)
df_sorted = df_clean.sort_values(by='Average_FSI_Total_Score', ascending=False)
midpoint = len(df_sorted) // 2

# 2. Split into Groups
# Top half = High Fragility (e.g., Somalia, Yemen)
group_fragile = df_sorted.head(midpoint)['Frequency']
# Bottom half = Low Fragility/Stable (e.g., Finland, Norway)
group_stable = df_sorted.tail(midpoint)['Frequency']

# 3. Perform Welch's T-Test (Equal Variance = False)
t_stat, p_value_t = ttest_ind(group_fragile, group_stable, equal_var=False)

# 4. Output Results
print("--- Welch's T-Test Results (High Fragility vs. Low Fragility) ---")
print(f"Mean Frequency (Fragile Group): {group_fragile.mean():,.0f}")
print(f"Mean Frequency (Stable Group):  {group_stable.mean():,.0f}")
print(f"T-Statistic: {t_stat:.3f}")
print(f"P-Value: {p_value_t:.5f}")

if p_value_t < 0.05:
    print("CONCLUSION: Statistically SIGNIFICANT difference.")
else:
    print("CONCLUSION: No significant difference between the groups.")

--- Welch's T-Test Results (High Fragility vs. Low Fragility) ---
Mean Frequency (Fragile Group): 15,896
Mean Frequency (Stable Group):  19,143
T-Statistic: -0.953
P-Value: 0.34229
CONCLUSION: No significant difference between the groups.


## Conclusion

Based on the statistical tests performed across all four indicators, we can rank the drivers of global news coverage by their correlation strength (Spearman's $\rho$):

1.  **Military Power:** Strongest predictor ($\rho \approx 0.65$, $p < 0.001$).
2.  **Economic Wealth (GDP):** Strong predictor ($\rho \approx 0.61$, $p < 0.001$).
3.  **State Fragility (FSI):** Moderate **negative** predictor ($\rho \approx -0.33$, $p < 0.001$).
    * *Interpretation:* As fragility increases (country becomes more unstable), news coverage tends to **decrease**. This contradicts the "if it bleeds, it leads" assumption for country-level data.
4.  **Democratic Status:** Weakest predictor ($\rho \approx 0.26$).

### Final Verdict: "Power Bias"
The data strongly supports the **"Power Bias"** hypothesis over the "Conflict Bias" hypothesis.
* Global news attention is primarily driven by **Hard Power** (Military and Economic strength).
* **Stability** is preferred over Fragility: highly stable, powerful nations receive significantly more attention than fragile, conflict-prone states (unless those states are strategically important).